File: Typescript/FlyweightPattern.js
var Character = /** @class */ (function () {
    function Character(character) {
        // Initialize a character object with the given character
        this.character = character;
    }
    Character.prototype.render = function (font, size) {
        // Simulate rendering the character with the provided font and size
        console.log("Character '".concat(this.character, "' rendered with font '").concat(font, "' and size ").concat(size));
    };
    return Character;
}());
var CharacterFactory = /** @class */ (function () {
    function CharacterFactory() {
    }
    CharacterFactory.getCharacter = function (character) {
        // Static method to get or create a character instance
        if (!CharacterFactory.characters[character]) {
            // If the character doesn't exist in the dictionary, create a new instance
            CharacterFactory.characters[character] = new Character(character);
        }
        // Return the character instance
        return CharacterFactory.characters[character];
    };
    CharacterFactory.characters = {}; // Dictionary to store created character instances
    return CharacterFactory;
}());
var TextEditor = /** @class */ (function () {
    function TextEditor() {
        this.characters = [];
    }
    TextEditor.prototype.addCharacter = function (character, font, size) {
        // Add a character to the text with the specified font and size
        var characterObj = CharacterFactory.getCharacter(character);
        this.characters.push([characterObj, font, size]);
    };
    TextEditor.prototype.render = function () {
        // Render all characters in the text
        for (var _i = 0, _a = this.characters; _i < _a.length; _i++) {
            var _b = _a[_i], character = _b[0], font = _b[1], size = _b[2];
            character.render(font, size);
        }
    };
    return TextEditor;
}());
// Example usage
var editor = new TextEditor();
// Add characters to the text with different fonts and sizes
editor.addCharacter("A", "Arial", 12);
editor.addCharacter("B", "Times New Roman", 14);
editor.addCharacter("A", "Arial", 12); // Reusing 'A' character
// Render the text
editor.render();

File: Python/FlyweightPattern.py
class Character:
    def __init__(self, character):
        # Initialize a character object with the given character
        self.character = character

    def render(self, font, size):
        # Simulate rendering the character with the provided font and size
        print(f"Character '{self.character}' rendered with font '{font}' and size {size}")


class CharacterFactory:
    _characters = {}  # Dictionary to store created character instances

    def get_character(character):
        # Static method to get or create a character instance
        if character not in CharacterFactory._characters:
            # If the character doesn't exist in the dictionary, create a new instance
            CharacterFactory._characters[character] = Character(character)
        # Return the character instance
        return CharacterFactory._characters[character]


class TextEditor:
    def __init__(self):
        # Initialize the TextEditor with an empty list to store characters
        self.characters = []

    def add_character(self, character, font, size):
        # Add a character to the text with the specified font and size
        character_obj = CharacterFactory.get_character(character)
        self.characters.append((character_obj, font, size))

    def render(self):
        # Render all characters in the text
        for character, font, size in self.characters:
            character.render(font, size)


# Example usage
editor = TextEditor()

# Add characters to the text with different fonts and sizes
editor.add_character('A', 'Arial', 12)
editor.add_character('B', 'Times New Roman', 14)
editor.add_character('A', 'Arial', 12)  # Reusing 'A' character

# Render the text
editor.render()
